src/browser/web-viewer.js

```diff
@@ -101,9 +118,31 @@ define((require, exports, module) => {
   WebViewer.onTitleChange = webViewerCursor => event =>
     webViewerCursor.set('title', event.detail);
 
-  WebViewer.onLocationChange = webViewerCursor => event =>
-    webViewerCursor.merge(Object.assign({location: event.detail},
-                                        getHardcodedColors(event.detail)));
+  const fetchThumbnail = uri => new Promise((resolve, reject) => {
+    const request = new XMLHttpRequest();
+    request.open('GET', `/tiles/${url.getDomainName(uri)}.png`);
+    request.responseType = 'blob';
+    request.send();
+    request.onload = event => resolve(request.response);
+    request.onerror = event => reject()
+  });
+
+  WebViewer.onThumbnailChanged = webViewerCursor => blob =>
+    webViewerCursor.set('thumbnail', URL.createObjectURL(blob));
+
+  const fetchScreenshot = iframe => new Promise((resolve, reject) => {
+    const request = iframe.getScreenshot(100, 62.5, 'image/png');
```

Comment:
> Isn't request a promise already? Test request.then.

I wish, it's [DOMRequest](https://developer.mozilla.org/en-US/docs/Web/API/DOMRequest)